frontend/js/behaviors/layeredImageViewer/layeredImageViewer.js

```diff
@@ -332,7 +457,131 @@ import OpenSeadragon from '../../libs/openseadragon4.min';
     // Reset
     this.toolbar.buttons.reset.addEventListener('click', () => {
       this.viewer.viewport.goHome();
+
+      // Deactivate annotations
+      // (Requires events to be dispatched manually to trigger)
+      const changeEvent = new Event('change');
+      this.annotations.items.forEach((annotation) => {
+        annotation.checkboxEl.checked = false;
+        annotation.checkboxEl.dispatchEvent(changeEvent);
+      });
     });
+
+    // Fullscreen
+    this.toolbar.buttons.fullscreen.addEventListener('click', () => {
+      if (!this.isFullscreen) {
+        this._setFullscreen(true);
+      } else {
+        this._setFullscreen(false);
+      }
+    });
+
+    // Control panel for annotations if present
+    this._addAnnotationMenu();
+  }
+
+  /**
+   * Add a menu for toggling annotations
+   * @private
+   * @method
+   */
+  _addAnnotationMenu() {
+    if (!this.annotations.items.length) return;
+
+    // Create template markup
+    const detailsTemplate = document.createElement('template');
+    detailsTemplate.innerHTML = `
+      <details class="layered-image-viewer-details">
+        <summary>Show annotations</summary>
+        <div class="layered-image-viewer-details__menu"></div>
+      </details>`;
+    const detailsEl = detailsTemplate.content.firstElementChild;
+    const panelEl = detailsEl.lastElementChild;
+
+    // Build up markup for the annotations panel
+    // Each item becoming a checkbox / label combo
+    const fieldTemplate = document.createElement('template');
+    this.annotations.items.forEach((item, i) => {
+      fieldTemplate.innerHTML = `
+        <div class="layered-image-viewer-details__option">
+          <input id="layered-image-viewer-${this.id}-annotation-cb-${i}" type="checkbox" data-index="${i}" />
+          <label for="layered-image-viewer-${this.id}-annotation-cb-${i}">${item.label}</label>
+        </div>
+        `;
```

Comment:
In contrast with my above comment, I understand why this HTML is here. Because you're adding elements to the OSD interface based on the number of entries in the CMS. I imagine this might be the only place this can happen.